### PR TITLE
Add SurveyJob Support in Foreman

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number

#715 

## Purpose/Implementation Notes

Adds SurveyJob Support in Foreman. Pretty simple implementation, uses the existing pattern.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests
New test

### CODE REVIEW

File: foreman/data_refinery_foreman/foreman/main.py [hunk 1]
```diff
@@ -391,6 +407,182 @@ def retry_lost_processor_jobs() -> None:
...
+        logger.info("Requeuing SurveyJob which had ID %d with a new SurveyJob with ID %d.",
+                    last_job.id,
+                    new_job.id)
+        try:
+            nomad_host = get_env_variable("NOMAD_HOST")
+            nomad_port = get_env_variable("NOMAD_PORT", "4646")
+            nomad_client = Nomad(nomad_host, port=int(nomad_port), timeout=5)
+            if dispatch:
+                nomad_response = nomad_client.job.dispatch_job("SURVEYOR", meta={"JOB_NAME": "SURVEYOR",
+                                                                                "JOB_ID": str(job.id)})
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/main.py]: Maybe we should switch surveyor jobs over to be expecting these meta values, however they are not currently what is accepted by surveyor jobs. They just need an accession.

**Comment by Miserlou** [comment]
Can you take another look at this? Has support for RAM increasing on surveyors now.

File: foreman/data_refinery_foreman/foreman/main.py [hunk 2]
```diff
@@ -393,6 +409,187 @@ def retry_lost_processor_jobs() -> None:
...
+
+                nomad_response = nomad_client.job.dispatch_job("SURVEYOR_" + ram_amount, meta={"JOB_ID": new_job.id})
+                new_job.nomad_job_id = nomad_response["DispatchedJobID"]
+                new_job.save()
+        except URLNotFoundNomadException:
+            logger.error("Dispatching Nomad job of type %s for job spec %s to host %s and port %s failed.",
+                         job_type, nomad_job, nomad_host, nomad_port, job=str(new_job.id))
+        except Exception as e:
+            logger.exception('Unable to Dispatch Nomad Job.',
+                job_name=job_type.value,
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/main.py]: `job_type` is undefined

Comment (kurtwheeler) [reply foreman/data_refinery_foreman/foreman/main.py]: As is nomad_job

File: foreman/data_refinery_foreman/surveyor/management/commands/surveyor_dispatcher.py [hunk 3]
```diff
@@ -13,28 +13,77 @@
...
+
+    set_source_type_for_accession(survey_job, accession)
+
     try:
-        nomad_response = nomad_client.job.dispatch_job("SURVEYOR", meta={"ACCESSION": accession})
+        nomad_response = nomad_client.job.dispatch_job("SURVEYOR", meta={"JOB_ID": survey_job.id})
     except URLNotFoundNomadException:
         logger.error("Dispatching Surveyor Nomad job to host %s and port %s failed.",
-                     job_type, nomad_job, nomad_host, nomad_port, accession_code=accession)
+                     SURVEYOR_JOB_NAME, nomad_job, nomad_host, nomad_port, accession_code=accession)
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/surveyor/management/commands/surveyor_dispatcher.py]: `nomad_job` is undefined.

File: foreman/data_refinery_foreman/surveyor/management/commands/surveyor_dispatcher.py [hunk 4]
```diff
@@ -13,28 +13,77 @@
...
     except Exception as e:
         logger.exception('Unable to Dispatch Nomad Job.',
-            job_name=job_type.value,
+            job_name=SURVEYOR_JOB_NAME,
             job_id=str(job.id),
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/surveyor/management/commands/surveyor_dispatcher.py]: `job` is undefined.

**Comment by kurtwheeler** [comment]
I think `dev` needs to be pulled into this branch.

**Comment by kurtwheeler** [comment]
From the staging test:
```
data-refinery-log-group-circleci-staging log-stream-foreman-circleci-staging 2018-11-02 14:08:10,503 i-0425dc317a979341e data_refinery_foreman.foreman.main ERROR [survey_job: 89]: Couldn't query Nomad about SurveyJob Job.
data-refinery-log-group-circleci-staging log-stream-foreman-circleci-staging Traceback (most recent call last):
data-refinery-log-group-circleci-staging log-stream-foreman-circleci-staging   File "/home/user/data_refinery_foreman/foreman/main.py", line 518, in retry_hung_survey_jobs
data-refinery-log-group-circleci-staging log-stream-foreman-circleci-staging     job_status = nomad_client.job.get_job(job.nomad_job_id)["Status"]
data-refinery-log-group-circleci-staging log-stream-foreman-circleci-staging AttributeError: 'SurveyJob' object has no attribute 'nomad_job_id'
data-refinery-log-group-circleci-staging log-stream-foreman-circleci-staging 2018-11-02 14:08:10,503 i-0425dc317a979341e data_refinery_foreman.foreman.main ERROR [survey_job: 90]: Couldn't query Nomad about SurveyJob Job.
```

You never ran half the code in this PR.... dude, you gotta start testing your stuff better

**Pull request merged**